### PR TITLE
[Where Clause] Support where clauses on ExprStmt [4/n]

### PR DESCRIPTION
**Pull request by mgrang**
We add support for where clauses on ExprStmts like:
  ```
  int a;
  a = 0 _Where a < 1 _And a != 2;
  ```

### CODE REVIEW

File: clang/lib/Parse/ParseStmt.cpp [hunk 1]
```diff
@@ -496,9 +496,28 @@ StmtResult Parser::ParseExprStatement(ParsedStmtContext StmtCtx) {
...
+    WClause = ParseWhereClause();
+    if (!WClause)
+      return StmtError();
+  }
+
   // Otherwise, eat the semicolon.
   ExpectAndConsumeSemi(diag::err_expected_semi_after_expr);
-  return handleExprStmt(Expr, StmtCtx);
+  StmtResult StmtRes = handleExprStmt(Expr, StmtCtx);
+  if (StmtRes.isInvalid() || !WClause)
```
Comment (sulekhark) [inline clang/lib/Parse/ParseStmt.cpp:510]: The check `!WClause` may not be necessary - it has already been checked above.

Comment (mgrang) [reply clang/lib/Parse/ParseStmt.cpp:510]: The `!WClause` check above is inside `if (getLangOpts().CheckedC && Tok.is(tok::kw__Where))`. If we are not in a where clause then we don't want the rest of the code below to execute. That's why I have this check here.

To make the logic cleaner, we can also have a dedicated variable which checks if where clause is present or not.

Comment (sulekhark) [reply clang/lib/Parse/ParseStmt.cpp:510]: Yes, thank you for the clarification!

**Review by sulekhark — APPROVED** [review]
LGTM!

**Comment by mgrang** [comment]
Closed in favor of https://github.com/microsoft/checkedc-clang/pull/1007

**Pull request closed without merging**